### PR TITLE
Add methods to ignore players when skipping the night

### PR DESCRIPTION
**Pull request by caseif**
This PR adds the API methods `isSleepingIgnored()Z` and `setSleepingIgnored(Z)V` to the `Player` interface. When this toggle is set, the `Player` will not be considered when determining whether the night should be skipped due to all players being asleep.

These methods are implemented by [SpongeCommon #74](https://github.com/SpongePowered/SpongeCommon/pull/74).

### CODE REVIEW

**Comment by simon816** [comment]
This should be on `Player`, because humans (like other creatures) don't affect the world sleep. The sleep is only for players.

File: src/main/java/org/spongepowered/api/entity/player/Player.java [hunk 1]
```diff
@@ -134,4 +134,26 @@
      */
     GameModeData getGameModeData();
 
+    /**
+     * Returns whether this {@link Player} will be ignored when checking whether
```
Comment (modwizcode) [inline src/main/java/org/spongepowered/api/entity/player/Player.java]: Gets

File: src/main/java/org/spongepowered/api/entity/player/Player.java [hunk 2]
```diff
@@ -134,4 +134,26 @@
...
+     */
+    boolean isSleepingIgnored();
+
+    /**
+     * Sets whether this {@link Player} will be ignored when checking whether
+     * to skip the night due to players sleeping. The time in a world will be
+     * advanced to day if all players in it either are sleeping or have this
+     * tag.
+     *
+     * @return Whether this {@link Player} will be ignored when checking whether
```
Comment (modwizcode) [inline src/main/java/org/spongepowered/api/entity/player/Player.java]: This is a parameter not a return

File: src/main/java/org/spongepowered/api/entity/player/Player.java [hunk 3]
```diff
@@ -134,4 +134,26 @@
...
     GameModeData getGameModeData();
 
+    /**
+     * Returns whether this {@link Player} will be ignored when checking whether
+     * to skip the night due to players sleeping. The time in a world will be
+     * advanced to day if all players in it either are sleeping or have this
+     * tag.
+     *
+     * @return Whether this {@link Player} will be ignored when checking whether
+     *         to skip the night
```
Comment (gabizou) [inline src/main/java/org/spongepowered/api/entity/player/Player.java]: indentation should be 4, not matching the start of the sentence.

like so:

```java
    /* ...
     * @return Whether this {@link Player} will be ignored when checking whether
     *     to skip the night
     */

```

**Comment by modwizcode** [comment]
Seems good.

**Pull request closed without merging**